stories/core/Select.stories.tsx

```diff
@@ -302,3 +302,58 @@ Custom.args = {
     { value: 'red', label: 'Red' },
   ],
 };
+
+export const MiddleTextTruncation: Story<SelectProps<string>> = (args) => {
+  const {
+    options = [
+      {
+        value:
+          'MyFileWithAReallyLongNameThatWillBeTruncatedBecauseItIsReallyThatLongSoHardToBelieve_FinalVersion_V2.html',
+        label:
+          'MyFileWithAReallyLongNameThatWillBeTruncatedBecauseItIsReallyThatLongSoHardToBelieve_FinalVersion_V2.html',
+      },
+      { value: 'ShortNameFile.jpg', label: 'ShortNameFile.jpg' },
+      { value: 'SomeOtherFile.dgn', label: 'SomeOtherFile.dgn' },
+    ],
+    placeholder = 'Placeholder text',
+    ...rest
+  } = args;
+  const [selectedValue, setSelectedValue] = useState<string | undefined>(
+    undefined,
+  );
+  return (
+    <div style={{ minHeight: 350, width: 300 }}>
+      <Select<string>
+        options={options}
+        value={selectedValue}
+        onChange={setSelectedValue}
+        placeholder={placeholder}
+        itemRenderer={(option) => (
+          <MenuItem>
+            <TextTruncation text={option.label} />
+          </MenuItem>
+        )}
+        selectedItemRenderer={(option) => (
+          <TextTruncation text={option.label} />
+        )}
+        {...rest}
+      />
+    </div>
+  );
+};
+
+MiddleTextTruncation.args = {
+  placeholder: 'Placeholder text',
```

Comment:
Why do these values need to be set twice?

Reply:
There is this issue with storybook.. It is going to override with undefined values as we take rest from the args..
Good catch, I moved rest above our inner set props and removed all the args